packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts

```diff
@@ -59,7 +59,9 @@ function deepCyclicCopyObject<T>(object: T, cycles: WeakMap<any, any>): T {
       descriptor.value = deepCyclicCopyReplaceable(descriptor.value, cycles);
     }
 
-    if (!('set' in descriptor)) {
+    if ('set' in descriptor) {
+      descriptor.set = undefined;
+    } else {
       descriptor.writable = true;
```

Comment:
This should be outside a condition? You could have a frozen object with a setter on it, right?

Reply:
We cannot set `writable` if there's a getter or a setter. So thanks for the question, we still failed if there was a getter but no setter 🙂 fixed and added a test

```js
Object.defineProperty({}, 'foo', {set(_){}, writable: true}) // Invalid property descriptor. Cannot both specify accessors and a value or writable attribute
Object.defineProperty({}, 'foo', {get(){}, writable: true}) // Invalid property descriptor. Cannot both specify accessors and a value or writable attribute
```

Reply:
Or did you have another example in mind?

Reply:
Added some more tests. With both getter and setter, only getter and only setter, both wrapped in `Object.freeze` and not.